api/db/database-builder/factory/build-authentication-method.js

```diff
@@ -4,7 +4,9 @@ const buildUser = require('./build-user');
 const AuthenticationMethod = require('../../../lib/domain/models/AuthenticationMethod');
 const encrypt = require('../../../lib/domain/services/encryption-service');
 
-const buildAuthenticationMethod = function ({
+const buildAuthenticationMethod = {};
+
+buildAuthenticationMethod.withGarAuthenticationComplement = function ({
```

Comment:
Je ne sais pas trop quoi penser de ce nom.
On utilise les `AuthenticationComplements` quand le reste des données n'est pas suffisant pour authentifier la personne via cette méthode.
Mais dans le cas du GAR on n'a justement aucun complément.

S'il fallait proposer un nom plus clair ça serait par exemple
```suggestion
buildAuthenticationMethod.withGarAsIdentityProvider = function ({
```

Et du coup pareil avec d'éventuels `withPixAsIdentityProvider` ou `withPoleEmploiAsIdentityProvider`

Reply:
Ah oui, j'ai mal compris le contenu du model d'AuthentificationMethod, j'ai en effet mélangé identityProviders et AuthenticationComplements !